lib/deterministicmnlist/SimplifiedMNList.js

```diff
@@ -333,4 +372,52 @@ SimplifiedMNList.prototype.toSimplifiedMNListDiff = function toSimplifiedMNListD
   }, network);
 };
 
+/**
+ * Deterministically selects all members of the quorum which
+ * has started it's DKG session with the block of this MNList
+ * @param {Buffer} selectionModifier
+ * @param {number} size
+ * @return {SimplifiedMNListEntry[]}
+ */
+SimplifiedMNList.prototype.calculateQuorum = function calculateQuorum(selectionModifier, size) {
+
+  var scores = this.calculateScores(selectionModifier);
+
+  scores.sort(function (a, b) {
+    return Buffer.compare(a.score, b.score);
+  });
+
+  scores.reverse();
+
+  return scores.map(function (score) {
+    return score.mn;
+  }).slice(0, size);
+};
+
+/**
+ * Calculates scores for MN selection
+ * it calculates sha256(sha256(proTxHash, confirmedHash), modifier) per MN
+ * Please note that this is not a double-sha256 but a single-sha256
+ * @param {Buffer} modifier
+ * @return {Object[]} scores
+ */
+SimplifiedMNList.prototype.calculateScores = function calculateScores(modifier) {
+
+  scores = [];
+  var allMns = this.getValidMasternodesList().slice();
+  allMns.forEach(function(mn) {
```

Comment:
you can use `validMNs.filter(function(mn) { return mn.confirmedHash; }).map( ... );`

Reply:
done
Also corrected the filter logic, because the property .confirmedHash cannot be missing, but it could theoretically be equal to nullHash